src/main/python/covata/delta/utils.py

```diff
@@ -34,3 +34,9 @@ def check_id(arguments):
     return check_arguments(arguments,
                            lambda x: x is not None and str(x) is not "",
                            "must be a nonempty string")
+
+
+def check_metadata(arguments):
+    return check_arguments(arguments,
+                           lambda x: x is not None and dict(x),
```

Comment:
We probably want to allow empty dictionary don't we? I need to revisit the logic because add_metadata only does strict add or update, does not remove.